packages/react-storage/src/components/StorageBrowser/createStorageBrowser.tsx

```diff
@@ -53,9 +53,9 @@ import {
  */
 export function createStorageBrowser<
   Input extends CreateStorageBrowserInput,
-  RInput extends Input['actions'] extends ExtendedActionConfigs
-    ? Input['actions']
-    : ExtendedActionConfigs,
+  RInput extends ExtendedActionConfigs & {
+    custom?: Input['actions'] extends {} ? Input['actions']['custom'] : {};
+  },
```

Comment:
Is this something we can address in `ExtendedActionConfigs` to reduce the amount of type inference logic we have here?

Reply:
We can. I write this logic to keep the `RInput` interface. However I don't think `RInput` generic is best for inference here. The better generic should be based on `Custom extends CustomActionConfigs`. Would changing this generic consider as breaking change.